actors/builtin/power/power_actor.go

```diff
@@ -421,6 +426,8 @@ func (a Actor) processBatchProofVerifies(rt Runtime) error {
 		rt.Abortf(exitcode.ErrIllegalState, "failed to batch verify: %s", err)
 	}
 
+	confirms := make([]*minerConfirmSector, 0, len(miners))
```

Comment:
I don't understand why this change is necessary. Could you please motivate it (and probably also in a comment for this structure)?

Reply:
@anorth 

**Earlier:** If a call to `BatchVerifyProofs` fails, all the sends to the `ConfirmSectorProofsValid` before that are successful and we abort which means there will be no further sends to `ConfirmSectorProofsValid` even for valid sectors/miners.

**Now:** We do ALL sends to `ConfirmSectorProofsValid` ONLY after we've verified all the sectors for all miners. 

I thought this behaviour would be easy to reason about. Let me know if this does not make sense and we want to keep the original behaviour and I'll make the change.

Reply:
Are you talking about the `Abortf` at line 434 here, if `m` is not found in the result from `BatchVerifySeals`. If we do abort there, all state changes are rolled back anyway, including those from any calls to `ConfirmSectorProofsValid` that happened earlier. An abort means that state will roll back as if the whole method call (including any subcalls) never happened. So as far as I can tell, this change makes no difference to the semantics, except costing less gas in an error case.

But also, if we reach that abort case, something is quite broken. It's not just a proof failing to verify.